### ISSUE TITLE
Style/MultilineArrayBraceLayout breaks code

### ISSUE THREAD

**Issue by ptarjan** [bbatsov/rubocop#3065]
## Expected behavior

```
$ cat a.rb
[foo,
 bar,
]
$ cat .rubocop.yml
Style/TrailingCommaInLiteral:
  EnforcedStyleForMultiline: comma
$ rubocop a.rb -a
Inspecting 1 file
E

Offenses:

a.rb:1:1: C: [Corrected] Style/MultilineArrayBraceLayout: Closing array brace must be on the same line as the last array element when opening brace is on the same line as the first array element.
[foo, ...
^^^^^

1 file inspected, 1 offenses detected, 1 offense corrected
$ cat a.rb
[foo,
 bar]
```

## Actual behavior

```
$ cat a.rb
[foo,
 bar,
]
$ cat .rubocop.yml
Style/TrailingCommaInLiteral:
  EnforcedStyleForMultiline: comma
$ rubocop a.rb -a
Inspecting 1 file
E

Offenses:

a.rb:1:1: C: [Corrected] Style/MultilineArrayBraceLayout: Closing array brace must be on the same line as the last array element when opening brace is on the same line as the first array element.
[foo, ...
^^^^^
a.rb:2:6: E: unexpected token tCOMMA
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
 bar],
     ^

1 file inspected, 2 offenses detected, 1 offense corrected
$ cat a.rb
[foo,
 bar],
```

## Steps to reproduce the problem

See my `a.rb`

## RuboCop version

```
$ rubocop -V
0.39.0 (using Parser 2.3.0.7, running on ruby 2.2.4 x86_64-darwin15)
```

**Comment by alexdowad** [comment]
Expected behavior = actual behavior on `0.40.0`. Please close this issue.

**Comment by ptarjan** [comment]
@alexdowad updated with a simple `rubocop.yml` that fails

**Comment by jonas054** [comment]
It's fixed on `master`, see #3152.
